solidity/contracts/GasRouter.sol

```diff
@@ -0,0 +1,97 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+pragma solidity >=0.6.11;
+
+import {Router} from "./Router.sol";
+
+abstract contract GasRouter is Router {
+    // ============ Mutable Storage ============
+    mapping(uint32 => uint256) public handleGasOverhead;
```

Comment:
I'm wondering if we should pick a different name here, particularly around "handle" & "overhead"

My reasoning is that it's only handle gas if the contract inheriting from `GasRouter` happens to be using the DefaultIsmInterchainGasPaymaster. If they happen to be using the base InterchainGasPaymaster, this doesn't quite apply

and then overhead feels like it's being used slightly differently from when I've been using it - I've been thinking of it as some kind of gas cost that's intrinsic and on top of some other application-specific gas. In this situation, this feels like "overhead" is used to refer to the application-specific gas

maybe something like `destinationGas` would be more general?